src/services/codefixes/importFixes.ts

```diff
@@ -235,6 +238,47 @@ namespace ts.codefix {
         }
     }
 
+    /**
+     * Computes module specifiers for multiple import additions to a file.
+     */
+     export interface ImportSpecifierResolver {

```

Comment:
I feel like I'm missing something obvious, but why not make this a function type?

Reply:
I really thought it was going to contain more than one method.

Reply:
To be fair in some languages they're the same. Just not in our language. :D

Reply:
So, I thought about moving more (all?) of the exported functions into here so that other things that want to call importFixes.ts logic don’t make the same mistake. E.g., I’m pretty sure the same inefficiency happens with the “add missing properties/methods” codefix when it needs to import more than one type to write out signatures. I decided against a bigger refactoring since we were considering this for a patch, but in the future I would like this object to hold all the importing-file-specific info involved with computing module specifiers that could get reused between resolving multiple module specifiers. That will also help with the growing issue in this file that every function takes about a dozen parameters that just get fed through from the top. It will be much easier to read these function signatures if we just close over `importingFile`, `program`, `host`, `preferences`, `formatContext`, `packageJsonImportFilter`, `exportInfoMap`, .... So that’s why I don’t want to replace this with `createGetModuleSpecifierForBestExportInfo` that just returns the one function.